src/components/login/SSOLogin.tsx

```diff
@@ -211,15 +235,73 @@ export default class SSOLogin extends Component<SSOLoginProps, SSOLoginState> {
     }
 
     checkConfigJson(ssoConfig) {
-        if (ssoConfig?.hasOwnProperty('clientID') && (ssoConfig?.clientID === DEFAULT_SECRET_PLACEHOLDER || !ssoConfig.clientID)) {
+        if (
+            ssoConfig?.hasOwnProperty('clientID') &&
+            (ssoConfig?.clientID === DEFAULT_SECRET_PLACEHOLDER || !ssoConfig.clientID)
+        ) {
             ssoConfig.clientID = ''
         }
-        if (ssoConfig?.hasOwnProperty('clientID') && (ssoConfig?.clientSecret === DEFAULT_SECRET_PLACEHOLDER || !ssoConfig.clientSecret)) {
+        if (
+            ssoConfig?.hasOwnProperty('clientID') &&
+            (ssoConfig?.clientSecret === DEFAULT_SECRET_PLACEHOLDER || !ssoConfig.clientSecret)
+        ) {
             ssoConfig.clientSecret = ''
         }
         return ssoConfig
     }
 
+    // The global auth config type needs to be updated irrespective of the SSO name check
+    _getGlobalAuthConfigType = () =>
+        !!AutoAssignToggleTile
+            ? {
+                  globalAuthConfigType:
+                      this.isAutoAssignPermissionFlowActive && this.state.shouldAutoAssignPermissions
+                          ? AUTHORIZATION_CONFIG_TYPES.GROUP_CLAIMS
+                          : AUTHORIZATION_CONFIG_TYPES.DEVTRON_MANAGED,
+              }
+            : {}
+
+    _getSSOCreateOrUpdatePayload = (configJSON) => ({
+        id: this.state.ssoConfig?.id,
```

Comment:
Either we should remove this optional check or add it on all the places if required